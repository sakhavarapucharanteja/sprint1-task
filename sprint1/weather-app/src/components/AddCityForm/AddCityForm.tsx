import { Box, TextField } from "@mui/material";
import React, { useState } from "react";

interface AddCityFormProps {
  handleAddCity: (cityName: string) => void;
}

const AddCityForm: React.FC<AddCityFormProps> = ({ handleAddCity }) => {
  const [value, setValue] = useState<string>("");

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    handleAddCity(value);
    setValue("");
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return (
    <Box
      onSubmit={handleSubmit}
      display="flex"
      justifyContent="space-between"
      component="form"
      autoComplete="off"
      noValidate
    >
      <TextField
        fullWidth
        onChange={handleChange}
        id="filled-basic"
        label="City"
        variant="filled"
        value={value}
      />
    </Box>
  );
};

export default AddCityForm;
