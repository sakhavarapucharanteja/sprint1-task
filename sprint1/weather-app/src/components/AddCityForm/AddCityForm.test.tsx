import { render, screen } from "@testing-library/react";
import AddCityForm from "./AddCityForm";

test("AddCityForm Renders", () => {
  render(
    <AddCityForm
      handleAddCity={(cityName: string): void => {
        throw new Error(cityName);
      }}
    />
  );
  const input = screen.getByLabelText("City");
  expect(input).toBeInTheDocument();
  expect(input).toMatchSnapshot();
});
