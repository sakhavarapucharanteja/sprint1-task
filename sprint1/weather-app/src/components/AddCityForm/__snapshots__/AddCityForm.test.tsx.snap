// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCityForm Renders 1`] = `
<input
  aria-invalid="false"
  class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
  id="filled-basic"
  type="text"
  value=""
/>
`;
